### PR TITLE
Upgrade to chartjs v4 and fix bugs

### PR DESCRIPTION
**Pull request by santam85**
Thanks for contributing.

- [X] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://developercertificate.org)

## Description

Upgraded the library to work with Chart.js v4. Change of configuration APIs was needed in order to avoid Chart.js errors.
Improved the testing examples in docs to showcase improved animations, color defaults matching the default Chart.js theme, updated code style, linting and packaging to match Chart.js standards.

## Testing

Check docs/index.html

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -1,35 +1,61 @@
...
+      - chartjs
+      - plugin:@typescript-eslint/recommended
+
+    rules:
+      complexity: ["warn", 10]
+      max-statements: ["warn", 30]
+      # Replace stock eslint rules with typescript-eslint equivalents for proper
+      # TypeScript support.
+      indent: "off"
+      "@typescript-eslint/indent": ["error", 2]
```
Comment (benmccann) [inline .eslintrc.yml:55]: there's a mix of formatting in this file. this rule is surrounded by double quotes and it's single quotes two lines below

File: docs/index.html [hunk 2]
```diff
@@ -1,51 +1,51 @@
...
-			</select>
-			<button id="update">Update</button>
-			<button id="randomizeData">Randomize Data</button>
-		</div>
-		<script type="text/javascript" src="index.js"></script>
-	</body>
+<head>
+	<meta charset="utf-8">
+	<meta name="viewport" content="width=device-width, initial-scale=1">
+	<title>Financial |  Chart.js</title>
```
Comment (benmccann) [inline docs/index.html:5]:
```suggestion
	<title>Financial | Chart.js</title>
```

File: docs/index.html [hunk 3]
```diff
@@ -1,51 +1,51 @@
...
+		<option value="false">No</option>
+	</select>
+	Mixed:
+	<select id="mixed">
+		<option value="true">Yes</option>
+		<option value="false" selected>No</option>
+	</select>
+	<button id="update">Update</button>
+	<button id="randomizeData">Randomize Data</button>
+</div>
```
Comment (benmccann) [inline docs/index.html:48]: I personally think it's a bit hard to read that this line and the next three all have the same level of indentation

File: src/controller.financial.js [hunk 4]
```diff
@@ -1,252 +1,156 @@
...
+  getUserBounds(scale) {
+    const {min, max, minDefined, maxDefined} = scale.getUserBounds();
+    return {
+      min: minDefined ? min : Number.NEGATIVE_INFINITY,
+      max: maxDefined ? max : Number.POSITIVE_INFINITY
+    };
+  }
+
+  /**
 	 * Implement this ourselves since it doesn't handle high and low values
```
Comment (benmccann) [inline src/controller.financial.js:88]: the formatting here looks messed up

**Review by benmccann — COMMENTED** [review]
there's really no way to review this with code changes and formatting changes mixed together. it probably would have been better to do the reformatting as a separate PR from the actual changes

**Comment by santam85** [comment]
> there's really no way to review this with code changes and formatting changes mixed together. it probably would have been better to do the reformatting as a separate PR from the actual changes

agreed, I jumped the gun a bit here, in the spirit of expediting alignment with the parent project.

**Comment by benmccann** [comment]
I'll leave it up to you then. Go ahead and merge if you feel confident enough in it. I don't think there's much help I can provide from looking at the diff

**Pull request merged**